### PR TITLE
Feat: support provider registry

### LINKED ISSUE #649: Support provider registry
1. Add data resource and a resource for a provider.
2. It should support changing the attributes the API allows.

See APIs here: https://docs.env0.com/reference/provider-registry-create-provider

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #649 

### Solution

1. Added API calls.
2. Added unit tests for new API calls.
3. Added provider registry resource.
4. Added acceptance tests for provider registry resource.
5. Added provider registry data source.
6. Added acceptance tests for provider registry data source.
7. Added integration tests for provider registry resource.
8. Added examples.

### CODE REVIEW

File: env0/resource_provider.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+		Importer: &schema.ResourceImporter{StateContext: resourceProviderImport},
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:             schema.TypeString,
+				Description:      `type of the provider registry (Match pattern: ^[0-9a-z](?:[0-9a-z-]{0,30}[0-9a-z])?$). Your provider’s type is essentially it’s name, and should match your provider’s files. For example, if your binaries look like terraform-provider-aws_1.1.1_linux_amd64.zip, than your provider’s type should be aws.`,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: NewRegexValidator(`^[0-9a-z](?:[0-9a-z-]{0,30}[0-9a-z])?$`),
```
Comment (TomerHeber) [inline env0/resource_provider.go:29]: I hope this regex is correct. I decided on this regex after "playing" with the UI.

File: env0/data_provider.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+func dataProvider() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataProviderRead,
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:         schema.TypeString,
+				Description:  "the type/name of the registry provider",
```
Comment (tomer-landesman) [inline env0/data_provider.go:18]: no need for the word 'registry', as this API is only for the providers themselves

File: env0/resource_provider.go (hunk 1 above)
Comment (tomer-landesman) [reply env0/resource_provider.go:29]: this is the regex we use in the UI 
`const PROVIDER_TYPE_INVALID_CHAR_REGEX = /^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$/;`

File: env0/resource_provider.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+	var foundProviders []client.Provider
+	for _, provider := range providers {
+		if provider.Type == name {
+			foundProviders = append(foundProviders, provider)
+		}
+	}
+
+	if len(foundProviders) == 0 {
+		return nil, fmt.Errorf("provider with name/type %v not found", name)
```
Comment (tomer-landesman) [inline env0/resource_provider.go:114]: lets stick with just `type` as this is the terminology we use with providers, so maybe also change the function name to `getProviderByType`

File: env0/resource_provider.go (hunk 1 above)
Comment (tomer-landesman) [reply env0/resource_provider.go:29]: we also have a max characters validation, we limit it to 32
`const MAX_CHARS = 32`

**Review by tomer-landesman — APPROVED** [review]
Looks good to me. left some small comments, mainly naming and error description changes

**Pull request merged**